### DISCUSSION TITLE
MDX Support

### DISCUSSION THREAD

**Discussion by tantaman** [Milkdown/milkdown#345] [Ideas]
Its a tall order -- but have you ever considered [MDX](https://mdxjs.com/) support?

Where you'd include the components from, and how, while editing a document is one of the first hurdles.

**Comment by Saul-Mirone** [comment]
It's absolutely possible since remark has already supports the [MDX plugin](https://www.npmjs.com/package/remark-mdx). The only problem is how can we handle the component it imports. Since milkdown is running on the browser, we need to find a way to import the files. That needs some investigation.

I'm still working on some exists TODO's in milkdown and vite-plugin-book. But I think support for MDX is something I really want to give it a try. Will do some investigation when I have time. If it's possible, it will be in our future plan.

**Reply by tantaman** [reply]
Nice. Its on my list of things to do for my own needs (although its pretty far down the list). If I get there first I'll send over some thoughts on how to handle imports and, if you agree on the approach, I'll send over some PRs.
